tc/core/check.cc

```diff
@@ -0,0 +1,39 @@
+#include "tc/core/check.h"
+namespace tc {
+namespace detail {
+
+Checker::Checker(bool condition, std::string location, std::string baseErrorMsg)
+    : condition_(condition), location_(location), baseErrorMsg_(baseErrorMsg){};
+
+Checker::~Checker() noexcept(false) {
+  if (condition_) {
+    return;
+  }
+  std::stringstream ss;
+  ss << "Check failed [" << location_ << ']';
+
+  if (not baseErrorMsg_.empty()) {
+    ss << ' ' << baseErrorMsg_;
+  }
+
+  if (not additionalMsg_.empty()) {
+    ss << ": " << additionalMsg_;
+  }
+  throw std::runtime_error(ss.str());
+}
+
+Checker tc_check(bool condition, const char* filename, uint64_t lineno) {
+  if (condition) {
+    return Checker(condition, {}, {});
+  }
```

Comment:
Why this special case?  Is it to avoid calling `makeLocation`?
Why not simply store `filename` and `lineno`?
Also, why does only this variant have such a special case?

Reply:
If the `condition` holds then the strings are not used, it's an "optimization". 

It was present in the rest of `tc_check_*` but I removed it because it didn't feel very useful. I forgot to remove this one.